File: web/src/types/market.ts
export type TokenizedCategory = 'stock' | 'gold' | 'silver' | 'etf' | 'tbill';

export interface TokenizedCategorySummary {
  category: TokenizedCategory;
  cmc_category_id: string;
  cmc_category_name: string;
  num_tokens: number | null;
  market_cap_usd: number | null;
  avg_price_change_24h: number | null;
  fetched_at: string;
}

export interface TokenizedAsset {
  id: string;
  symbol: string;
  name: string;
  category: TokenizedCategory;
  cmc_rank: number | null;
  price_usd: number | null;
  market_cap_usd: number | null;
  volume_24h_usd: number | null;
  percent_change_1h: number | null;
  percent_change_24h: number | null;
  percent_change_7d: number | null;
  network: string | null;
  last_updated: string | null;
}

export interface MarketSnapshot {
  summary: TokenizedCategorySummary[];
  assets: TokenizedAsset[];
  coverage_note: string;
  last_fetched: string | null;
}

File: backend/app/schemas/market.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, ConfigDict


class TokenizedAssetRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    symbol: str
    name: str
    category: str
    cmc_rank: Optional[int] = None
    price_usd: Optional[float] = None
    market_cap_usd: Optional[float] = None
    volume_24h_usd: Optional[float] = None
    percent_change_1h: Optional[float] = None
    percent_change_24h: Optional[float] = None
    percent_change_7d: Optional[float] = None
    network: Optional[str] = None
    last_updated: Optional[datetime] = None
    fetched_at: Optional[datetime] = None


class TokenizedCategorySummaryRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    category: str
    cmc_category_id: str
    cmc_category_name: str
    num_tokens: Optional[int] = None
    market_cap_usd: Optional[float] = None
    avg_price_change_24h: Optional[float] = None
    fetched_at: Optional[datetime] = None


class MarketSnapshot(BaseModel):
    summary: list[TokenizedCategorySummaryRead]
    assets: list[TokenizedAssetRead]
    coverage_note: str
    last_fetched: datetime
